### PR TITLE
Improve Option documentation and add MaybeSome function

### PR DESCRIPTION
**Pull request by ffranr**
Fix `Option` doc comment and add `MaybeSome` function for pointer conversion.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Thanks, LGTM :tada:

File: fn/option.go [hunk 1]
```diff
@@ -26,6 +26,17 @@ func None[A any]() Option[A] {
 	return Option[A]{}
 }
 
+// MaybeSome constructs an option from a pointer.
+//
+// MaybeSome : *A -> Option[A].
+func MaybeSome[A any](a *A) Option[A] {
```
Comment (ProofOfKeags) [inline fn/option.go:32]: This function is good. However, please rename it to something approximating the below list. "Maybe" does not have precedent to mean anything and I think there are more intuitive names for this signature.

- `OptionFromPtr`
- `PtrToOption`
- `DerefOrNone`

**Review by ProofOfKeags — CHANGES_REQUESTED** [review]

File: fn/option.go (hunk 1 above)
Comment (ffranr) [reply fn/option.go]: I've gone with `OptionFromPtr` in the latest commits.

Thanks for taking a look.

**Review by ProofOfKeags — APPROVED** [review]

**Comment by guggero** [comment]
I pushed a new tag `fn/v1.2.4` for this.

**Pull request merged**